Systemd oomd

rfrohl opened the pull request:
Wanted to get some feedback on this backlog change I have been working when there was time. I still have a couple of AVCs that I am not sure how to address:

> type=AVC msg=audit(..): avc:  denied  { search } for  pid=1326 comm="systemd-oomd" name="3209" dev="proc" ino=22552 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=dir permissive=1
> type=AVC msg=audit(..): avc:  denied  { read } for  pid=1326 comm="systemd-oomd" name="stat" dev="proc" ino=22553 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=file permissive=1
> type=AVC msg=audit(..): avc:  denied  { open } for  pid=1326 comm="systemd-oomd" path="/proc/3209/stat" dev="proc" ino=22553 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=file permissive=1
> type=AVC msg=audit(..): avc:  denied  { getattr } for  pid=1326 comm="systemd-oomd" path="/proc/3209/stat" dev="proc" ino=22553 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=file permissive=1
> type=AVC msg=audit(..): avc:  denied  { ioctl } for  pid=1326 comm="systemd-oomd" path="/proc/3209/stat" dev="proc" ino=22553 ioctlcmd=0x5401 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=file permissive=1

Open to pointers for these or any other feedback.

rfrohl committed:
Basic enablement for systemd-oomd

dontaudit dac_override:
type=AVC msg=audit(..): avc:  denied  { dac_override } for  pid=2643 comm="systemd-oomd" capability=1  scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:system_r:systemd_oomd_t:s0 tclass=capability permissive=1

allow systemd_oomd_t systemd_oomd_var_run_t:sock_file manage_sock_file_perms:
type=AVC msg=audit(..): avc:  denied  { write } for pid=4361 comm="systemd-oomd" name="io.systemd.ManagedOOM" dev="tmpfs" ino=42 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:systemd_oomd_var_run_t:s0 tclass=sock_file permissive=1

manage_sock_file_perms;
kernel_dgram_send(systemd_oomd_t):
type=AVC msg=audit(..): avc:  denied  { sendto } for  pid=2643 comm="systemd-oomd" path="/systemd/journal/socket" scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:system_r:kernel_t:s0 tclass=unix_dgram_socket permissive=1

kernel_read_psi(systemd_oomd_t):
type=AVC msg=audit(..): avc:  denied  { search } for  pid=2643 comm="systemd-oomd" name="pressure" dev="proc" ino=4026532057 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:proc_psi_t:s0 tclass=dir permissive=1
type=AVC msg=audit(..): avc:  denied  { read } for  pid=2643 comm="systemd-oomd" name="cpu" dev="proc" ino=4026532060 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:proc_psi_t:s0 tclass=file permissive=1
type=AVC msg=audit(..): avc:  denied  { open } for  pid=2643 comm="systemd-oomd" path="/proc/pressure/cpu" dev="proc" ino=4026532060 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:proc_psi_t:s0 tclass=file permissive=1
type=AVC msg=audit(..): avc:  denied  { getattr } for  pid=2643 comm="systemd-oomd" path="/proc/pressure/cpu" dev="proc" ino=4026532060 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:proc_psi_t:s0 tclass=file permissive=1

kernel_stream_connect(systemd_oomd_t):
type=AVC msg=audit(..): avc:  denied  { connectto } for  pid=2643 comm="systemd-oomd" path="/systemd/io.systemd.ManagedOOM" scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:system_r:kernel_t:s0 tclass=unix_stream_socket permissive=1

dbus_acquire_svc_system_dbusd(systemd_oomd_t):
type=USER_AVC msg=audit(..): pid=885 uid=496 auid=4294967295 ses=4294967295 subj=stystem_u:system_r:system_dbusd_t:s0-s0:c0.c1023 msg='avc:  denied  { acquire_svc } for  scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:system_r:system_dbusd_t:s0-s0:c0.c1023 tclass=dbus permissive=1 exe="/usr/bin/dbus-broker" sauid=496 hostname=? addr=? terminal=?'

Fixes: bsc#1236535
Signed-off-by: Robert Frohl <rfrohl@suse.com>

rfrohl committed:
Basic functionality for systemd-oomd

domain_read_all_domains_state(systemd_oomd_t):
type=AVC msg=audit(..): avc:  denied  { search } for  pid=1368 comm="systemd-oomd" name="24509" dev="proc" ino=80823 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=dir permissive=1
type=AVC msg=audit(..): avc:  denied  { read } for  pid=1368 comm="systemd-oomd" name="stat" dev="proc" ino=80824 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s
0-s0:c0.c1023 tclass=file permissive=1
type=AVC msg=audit(..): avc:  denied  { open } for  pid=1368 comm="systemd-oomd" path="/proc/24509/stat" dev="proc" ino=80824 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:un
confined_t:s0-s0:c0.c1023 tclass=file permissive=1
type=AVC msg=audit(..): avc:  denied  { getattr } for  pid=1368 comm="systemd-oomd" path="/proc/24509/stat" dev="proc" ino=80824 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=file permissive=1
type=AVC msg=audit(..): avc:  denied  { ioctl } for  pid=1368 comm="systemd-oomd" path="/proc/24509/stat" dev="proc" ino=80824 ioctlcmd=0x5401 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=file permissive=1

domain_kill_all_domains(systemd_oomd_t):
type=AVC msg=audit(..): avc:  denied  { kill } for pid=1433 comm="systemd-oomd" capability=5 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:system_r:systemd_oomd_t:s0 tclass=capability permissive=1
type=AVC msg=audit(..): avc:  denied  { sigkill } for pid=1433 comm="systemd-oomd" scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=process permissive=1

fs_manage_cgroup_dirs(systemd_oomd_t)
type=AVC msg=audit(..): avc:  denied  { read } for  pid=1333 comm="systemd-oomd" name="/" dev="cgroup2" ino=1 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(..): avc:  denied  { read } for  pid=1333 comm="systemd-oomd" name="background.slice" dev="cgroup2" ino=4594 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(..): avc:  denied  { write } for  pid=1333 comm="systemd-oomd" name="run-p1791-i1792.service" dev="cgroup2" ino=5024 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(..): avc:  denied  { setattr } for  pid=1333 comm="systemd-oomd" name="run-p1791-i1792.service" dev="cgroup2" ino=5024 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1

fs_list_cgroup_dirs(systemd_oomd_t):
type=AVC msg=audit(..): avc:  denied  { read } for pid=1433 comm="systemd-oomd" name="system.slice" dev="cgroup2" ino=73 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(..): avc:  denied  { read } for pid=1433 comm="systemd-oomd" name="session.slice" dev="cgroup2" ino=6774 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1

fs_setattr_cgroup_dirs(systemd_oomd_t):
type=AVC msg=audit(..): avc:  denied  { setattr } for pid=1433 comm="systemd-oomd" name="run-p6559-i6560.service" dev="cgroup2" ino=10300 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1

fs_write_cgroup_dirs(systemd_oomd_t):
type=AVC msg=audit(..): avc:  denied  { write } for pid=1433 comm="systemd-oomd" name="run-p6559-i6560.service" dev="cgroup2" ino=10300 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1

dbus_read_pid_sock_files(systemd_oomd_t):
audit(..): avc:  denied  { watch } for  pid=1480 comm="systemd-oomd" path="/run" dev="tmpfs" ino=1 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:var_run_t:s0 tclass=dir permissive=1

dbus_watch_pid_dir_path(systemd_oomd_t):
type=AVC msg=audit(..): avc:  denied  { watch } for  pid=1482 comm="systemd-oomd" path="/" dev="nvme0n1p2" ino=256 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:root_t:s0 tclass=dir permissive=1

Fixes: bsc#1236535
Signed-off-by: Robert Frohl <rfrohl@suse.com>

zpytela commented:
Thank you for the contribution. Before I start with the review, please answer briefly:
- What system environment, settings, scenarios did you use for testing?
- Have you also booted the system with "cgroup_disable=pressure"?
- Have you tried different reasonable settings of the ManagedOOM variables?
- Have you tried oomctl with the sysadm user?

rfrohl commented:
> * What system environment, settings, scenarios did you use for testing?

openSUSE Tumbleweed (systemd 257.6, probably with some patches on top). In addition I had to change:

```
# from /etc/systemd/system.conf.d/21-oomd.conf
[Manager]
# Override /usr/lib/systemd/system.conf.d/20-defaults-SUSE.conf
DefaultMemoryAccounting=true
```

Tested that: 

- the daemon is able to start
- the scenario documented under [EnableSystemdOomd#How_to_test](https://fedoraproject.org/wiki/Changes/EnableSystemdOomd#How_to_test) is working without AVCs
- ran 'oomctl dump'

> 
> * Have you also booted the system with "cgroup_disable=pressure"?

No, but I can try that as well.

> 
>  * Have you tried different reasonable settings of the ManagedOOM variables?

I only tried the scenario from How_to_test, but I do not mind trying something else as well. My main problem was how to trigger any of the scenarios.
 
> 
>  * Have you tried oomctl with the sysadm user?

I have not, but can do that tomorrow.

zpytela commented on policy/modules/system/systemd.te:
```diff
@@ -1904,6 +1909,36 @@ permissive systemd_mountfsd_t;
+# systemd_oomd local policy
+#
+
+dontaudit systemd_oomd_t self:capability dac_override;
```
For which path this capability check was requested?

rfrohl replied:
I hope that the answer comes via full auditing, as I am still new to selinux and not 100% sure if that assumption is correct: `name=/proc/1/environ`


full details below: (same test system, semodule -DB + full auditing):

```
----
type=PROCTITLE msg=audit(07/03/25 17:32:32.233:133) : proctitle=/usr/lib/systemd/systemd-oomd                                                                                                                       
type=PATH msg=audit(07/03/25 17:32:32.233:133) : item=0 name=/proc/1/environ inode=7982 dev=00:48 mode=file,400 ouid=root ogid=root rdev=00:00 obj=system_u:system_r:init_t:s0 nametype=NORMAL cap_fp=none cap_fi=none cap_fe=0 cap_fver=0 cap_frootid=0                 
type=CWD msg=audit(07/03/25 17:32:32.233:133) : cwd=/                                                                                                                                                               type=SYSCALL msg=audit(07/03/25 17:32:32.233:133) : arch=x86_64 syscall=openat success=no exit=EACCES(Permission denied) a0=AT_FDCWD a1=0x7fff9b6b8e00 a2=O_RDONLY|O_CLOEXEC a3=0x0 items=1 ppid=1 pid=1344 auid=uns
et uid=systemd-oom gid=systemd-oom euid=systemd-oom suid=systemd-oom fsuid=systemd-oom egid=systemd-oom sgid=systemd-oom fsgid=systemd-oom tty=(none) ses=unset comm=systemd-oomd exe=/usr/lib/systemd/systemd-oomd subj=system_u:system_r:systemd_oomd_t:s0 key=(null)  
type=AVC msg=audit(07/03/25 17:32:32.233:133) : avc:  denied  { dac_override } for  pid=1344 comm=systemd-oomd capability=dac_override  scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:system_r:syst
emd_oomd_t:s0 tclass=capability permissive=1
----
```

zpytela replied:
That does not feel right, needs to be checked further.

zpytela replied:
It is executed from detect_container(). Not sure what it would effect in if the permission is not allowed once the domain loses the permissive status.

rfrohl replied:
I did test the scenario with the domain in enforcing, I did not see a difference. But I wonder if I might miss certain functionality in my setup.

Re-testing this again with the changes currently, will report back if there are problems with the domain in enforcing.

rfrohl replied:
can confirm that on my test system, with the domain in enforcing, and the latest changes these work:

- oomd starts
- [this test case](https://fedoraproject.org/wiki/Changes/EnableSystemdOomd#How_to_test) works
- 'oomctl dump' returns data

zpytela commented on policy/modules/system/systemd.te:
```diff
@@ -1904,6 +1909,36 @@ permissive systemd_mountfsd_t;
+kernel_stream_connect(systemd_oomd_t)
+
+# needed for daemon operation
+files_watch_root_dirs(systemd_oomd_t)
```
I guess dbus_watch_pid_dir_path() would be a better choice for cases where multiple filesystems are in the path.

rfrohl replied:
will change that

zpytela commented on policy/modules/system/systemd.te:
```diff
@@ -1904,6 +1909,36 @@ permissive systemd_mountfsd_t;
+
+# needed for daemon operation
+files_watch_root_dirs(systemd_oomd_t)
+fs_manage_cgroup_dirs(systemd_oomd_t)
```
Are there any other cgroup files/dirs than mentioned in the commit message?

rfrohl replied:
there were a bit more, see the AVCs from back then below:

```
type=AVC msg=audit(1749717757.257:154): avc:  denied  { read } for  pid=1326 comm="systemd-oomd" name="user@0.service" dev="cgroup2" ino=4164 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1749717757.276:155): avc:  denied  { read } for  pid=1326 comm="systemd-oomd" name="app.slice" dev="cgroup2" ino=4250 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1749718907.125:156): avc:  denied  { read } for  pid=1333 comm="systemd-oomd" name="user@0.service" dev="cgroup2" ino=4207 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1749718907.126:157): avc:  denied  { read } for  pid=1333 comm="systemd-oomd" name="app.slice" dev="cgroup2" ino=4293 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1749718908.376:158): avc:  denied  { read } for  pid=1333 comm="systemd-oomd" name="app.slice" dev="cgroup2" ino=4293 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1749721377.624:271): avc:  denied  { read } for  pid=1333 comm="systemd-oomd" name="/" dev="cgroup2" ino=1 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1749721377.630:272): avc:  denied  { read } for  pid=1333 comm="systemd-oomd" name="background.slice" dev="cgroup2" ino=4594 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1749721377.644:273): avc:  denied  { write } for  pid=1333 comm="systemd-oomd" name="run-p1791-i1792.service" dev="cgroup2" ino=5024 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1749721377.644:274): avc:  denied  { setattr } for  pid=1333 comm="systemd-oomd" name="run-p1791-i1792.service" dev="cgroup2" ino=5024 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1749726756.440:305): avc:  denied  { write } for  pid=1333 comm="systemd-oomd" name="run-p3573-i3574.service" dev="cgroup2" ino=5712 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1
type=AVC msg=audit(1749726756.440:306): avc:  denied  { setattr } for  pid=1333 comm="systemd-oomd" name="run-p3573-i3574.service" dev="cgroup2" ino=5712 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=unconfined_u:object_r:cgroup_t:s0 tclass=dir permissive=1
```

rfrohl replied:
fixed I believe

zpytela replied:
I think read+write+setattr does not necessarily mean manage permissions are needed, I created 2 new interfaces.

rfrohl replied:
transitioned to the new interfaces

zpytela commented on policy/modules/system/systemd.te:
```diff
@@ -1904,6 +1909,36 @@ permissive systemd_mountfsd_t;
+domain_kill_all_domains(systemd_oomd_t)
+
+# needed for oomctl
+fs_rw_inherited_tmpfs_files(systemd_oomd_t)
```
Which file is it, created by which service?

rfrohl replied:
the file was '/memfd:data-fd (deleted)', not sure how I can check who created it.

zpytela replied:
If it was systemd-oomd, it should get a private type. That is also a way how to help with answering the who question.

rfrohl replied:
I followed the other hint, but was unsure if I should provide the interface as well already ? Just added the type for now.

zpytela commented on policy/modules/system/systemd.te:
```diff
@@ -1904,6 +1909,36 @@ permissive systemd_mountfsd_t;
+# needed for oomctl
+fs_rw_inherited_tmpfs_files(systemd_oomd_t)
+
+init_named_pid_filetrans(systemd_oomd_t, systemd_oomd_var_run_t, file, "oom")
```
A plain file is not mentioned in systemd.fc, so why is this needed?

rfrohl replied:
I do not recall, will check tomorrow: I assume for the folder, i.e. `/run/systemd/oom` and I did not change remove the `file` bit.

rfrohl replied:
I will remove this:

I think I thought that I needed a filetrans for the directory creation, but only added something half complete. Retesting without this bit works without issues. So I would drop it.

Sorry that this slipped through.

rfrohl replied:
fixed I believe

rfrohl commented:
> Have you also booted the system with "cgroup_disable=pressure"?

That seems to break systemd-oomd, as it still starts but does not do anything anymore. The kernel oom killer starts to kill processes as fallback then.

Saw no AVCs when testing this.

rfrohl commented:
I struggle a bit with this one, not 100% sure what 'the sysadm user' is.

> Have you tried oomctl with the sysadm user?

Tried 'oomctl dump' with a confined user, following the setup documented [here](https://docs.redhat.com/de/documentation/red_hat_enterprise_linux/8/html/using_selinux/managing-confined-and-unconfined-users_using-selinux#confining-an-administrator-using-sudo-and-the-sysadm_r-role_managing-confined-and-unconfined-users).

The user confined to staff_u is able to run the command without AVCs and gets the same results as root running it.

A user confined to sysadm_u I do not get to log in, not sure if I am trying something wrong.

zpytela commented:
Funny thing is that on my systems this is the needed policy:
```

allow systemd_oomd_t self:capability kill;
allow systemd_oomd_t systemd_oomd_var_run_t:sock_file write_sock_file_perms;

kernel_dgram_send(systemd_oomd_t)
kernel_read_psi(systemd_oomd_t)
kernel_stream_connect(systemd_oomd_t)

domain_read_all_domains_state(systemd_oomd_t)
domain_kill_all_domains(systemd_oomd_t)

fs_list_cgroup_dirs(systemd_oomd_t)
fs_setattr_cgroup_dirs(systemd_oomd_t)
fs_write_cgroup_dirs(systemd_oomd_t)

optional_policy(`
    dbus_acquire_svc_system_dbusd(systemd_oomd_t)
    dbus_read_pid_sock_files(systemd_oomd_t)
    dbus_watch_pid_sock_files(systemd_oomd_t)
    dbus_watch_pid_dir_path(systemd_oomd_t)
')
```
Not a subset, not a superset of yours. Even worse, on different systems of mine the permissions were not completely the same.

rfrohl commented:
>  Not a subset, not a superset of yours. Even worse, on different systems of mine the permissions were not completely the same.

Where does that leave us with this proposal ? oomd is still experimental, not sure if that might play a role. I switched to the two new interfaces, but not sure if the PR has a chance to get merged if there are three different set of rules needed on three systems. 

The two open points that are left I would also need some time to investigate, because I do not have a clear way for me on how to answer them. So would like to know if spending the time would be worth it or if there is another way to tackle it.

zpytela commented:
> > Not a subset, not a superset of yours. Even worse, on different systems of mine the permissions were not completely the same.
> 
> Where does that leave us with this proposal ? oomd is still experimental, not sure if that might play a role. I switched to the two new interfaces, but not sure if the PR has a chance to get merged if there are three different set of rules needed on three systems.
I' ll merge it at some point to move forward.

> 
> The two open points that are left I would also need some time to investigate, because I do not have a clear way for me on how to answer them. So would like to know if spending the time would be worth it or if there is another way to tackle it.

Not sure which two you mean: memfd should be addressed by using files_tmpfs_file (see e.g. 56619afc0f), about /proc/1/environ I am not sure yet and i suggest to postpone cgroup_disable problem.

rfrohl committed:
Allow 'oomctl dump' to interact with systemd-oomd

type=AVC msg=audit(..): avc:  denied  { write } for  pid=3357 comm="systemd-oomd" path=/memfd:data-fd (deleted) dev="tmpfs" ino=2092 scontext=system_u:system_r:systemd_oomd_t:s0 tcontext=system_u:object_r:tmpfs_t:s0 tclass=file permissive=1

Fixes: bsc#1236535
Signed-off-by: Robert Frohl <rfrohl@suse.com>

rfrohl commented:
> Not sure which two you mean: memfd should be addressed by using files_tmpfs_file (see e.g. https://github.com/fedora-selinux/selinux-policy/commit/56619afc0fb6eabca1d9ef8f8d552b21627322ff)

The two open discussions that are still unresolved. Thanks the tmpfs pointer helped !

zpytela commented:
I see only 1 unresolved issue now. I am fine with merging this PR once new userspace is publicly available in all environments and checks do not fail.

rfrohl commented:
I got to review user feedback, that has been very helpful with testing this. There are a few more AVCs that lead to something similar as in this comment:

>   Funny thing is that on my systems this is the needed policy:
> [..]
>    dbus_read_pid_sock_files(systemd_oomd_t)
>    dbus_watch_pid_sock_files(systemd_oomd_t)

I would add these two as well before merging.

**edit:** realized only now that *dbus_watch_pid_sock_files()* is a subset of *dbus_watch_pid_dir_path()*, so will just extend the proposal with **dbus_read_pid_sock_files()**

zpytela commented:
CI is not ready yet, but in the meantime I created a simple test.
https://src.fedoraproject.org/tests/selinux/pull-request/681

zpytela commented:
I am going to merge this PR if you don't mind.

rfrohl commented:
> I am going to merge this PR if you don't mind.

please go ahead, nothing missing from my current point of view.

zpytela commented:
Merging, thank you.
